crew

```diff
@@ -529,17 +529,31 @@ def download
 
   uri = URI.parse url
   filename = File.basename(uri.path)
+  
   if source
-    sha256sum = @pkg.source_sha256
+    if @pkg.source_sha512
```

Comment:
Will this throw an error if source_sha512 is not set?

Reply:
No, as this is a instance variable (return nil if not set)